stdlib/src/memory/maybe_uninitialized.mojo

```diff
@@ -0,0 +1,77 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+
+
+# TODO: Make this public when we are certain of the design.
+# TODO: Move _size into an alias when the bug https://github.com/modularml/mojo/issues/2889
+# is fixed.
+struct _MaybeUninitialized[ElementType: CollectionElement, _size: Int = 1](
```

Comment:
**Suggestion** I suggest we remove this `_size` parameter and always use a fixed-size of `1`. To model arrays of possibly-uninitialized elements, authors could use `InlineArray[MaybeUninitialized[T], size]`. That composes and generalizes a bit better I think that `MaybeUninitialized[T, size]`.

E.g. `List[MaybeUninitialized[T]]` would work just as well.

Reply:
See the comment just above, line 17. The `_size` here is not meant to be modified and we indeed wish to make it fixed. But we can't because of a bug. The issue has already been opened about this and the link provided in the comment.

Reply:
Understood. I wonder if if we use a literal `1` in the `pop.array` type, does that run into the same bug?